### DISCUSSION TITLE
These models do not have a unique identifier or id and are therefore commented out.

### DISCUSSION THREAD

**Discussion by firmanjml** [prisma/prisma#2669] [Q&A] [answered]
## Description

Not sure if this is a bug or not, that's why I post it here instead.

I attempted to try out the many to many self relationships with the instruction given on the docs but the output isn't the same in the doc.
 
It doesn't generate the model as I expected it to be.
Followed the tutorial on [Relations (Reference)](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#many-to-many-relations)

## How to reproduce

1. Go to the database
2. Change add the following sql 

`

    CREATE SEQUENCE "User_id_seq";

    CREATE TABLE "User" (
        id integer DEFAULT nextval('"User_id_seq"'::regclass) PRIMARY KEY,
        name text
     );

    CREATE TABLE "_UserFollows" (
        "A" integer NOT NULL REFERENCES "User"(id) ON DELETE CASCADE ON UPDATE CASCADE,
        "B" integer NOT NULL REFERENCES "User"(id) ON DELETE CASCADE ON UPDATE CASCADE
    );
`

3. Run npx prisma introspect
4. See warning message

## Expected behavior

Expected behavior should be as shown on the doc in [Relations (Reference)](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#many-to-many-relations)

`

    model User {
        id          Int      @id @default(autoincrement())
        name        String?
        followedBy  User[]   @relation("UserFollows")
        following   User[]   @relation("UserFollows")
    }

`

## Environment & setup

- OS: MacOS
- Database: PostgreSQL
- Prisma version:  2.0.0-beta.8
- Node.js version: v14.3.0

**Comment by janpio** [comment, marked as answer]
What is the schema that is generated for you besides the warning?

Generally you probably are missing some of the (strict) details that need to be fulfilled for these implicit tables to be picked up by Introspection. See the examples at https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#conventions-for-relation-tables-in-implicit-m-n-relations (You seem to be missing the indexes at least)

**Reply by firmanjml** [reply]
It will just generate User model.
Doesnt have self many-to-many relationships.

I tried adding an index in it but it's still different than the example.
With the index, it will generate UserFollow model unlike the example given in the doc, it will add @relation and points it out to _UserFollow with the User[] model.

If i add index in userfollow table, it will point to UserFollow[] model.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
